src/components/TransitionProperties.tsx

```diff
@@ -0,0 +1,213 @@
+import * as CheckBox from "grommet/components/CheckBox";
+import * as TextInput from "grommet/components/TextInput";
+import { hideTransitionProperties, setJsonMessageString, setCurrentId, setPrivateTopic } from "actions/transitionProperties";
+import Instances from "components/Instances";
+import { XCSpyState } from "reducers/SpyReducer";
+
+interface TransitionPropertiesGlobalProps extends TransitionPropertiesProps, TransitionPropertiesCallbackProps {
+};
+
+interface TransitionPropertiesProps {
+    privateTopic: string;
+    id: string;
+    jsonMessageString: string;
+    messageType: string;
+    active: boolean;
+    stateMachine: string;
+    currentComponent: string;
+    getStateMachineRefFromId: (id: string) => any;
+};
+
+interface TransitionPropertiesCallbackProps {
+    setPrivateTopic: (privateSend: string) => void;
+    setCurrentId: (id: string) => void;
+    setJsonMessageString: (jsonMessageString: string) => void;
+    hideTransitionProperties: () => void;
+    send: (component: string, stateMachine: string, messageType: string, jsonMessageString: string) => void;
+    sendContext: (stateMachineRef: any, messageType: string, jsonMessageString: string) => void;
+};
+
+
+const mapStateToProps = (state: XCSpyState): TransitionPropertiesProps => {
+    return {
+        privateTopic: state.transitionProperties.privateTopic,
+        id: state.transitionProperties.id,
+        jsonMessageString: state.transitionProperties.jsonMessageString,
+        messageType: state.transitionProperties.messageType,
+        active: state.transitionProperties.active,
+        stateMachine: state.transitionProperties.stateMachine,
+        currentComponent: state.components.currentComponent,
+        getStateMachineRefFromId: (id) => {
+            if (!id) {
+                return null;
+            }
+            let componentProperties = state.components.componentProperties;
+            let currentComponent = state.components.currentComponent;
+            let stateMachine = state.transitionProperties.stateMachine;
+            let instance = componentProperties[currentComponent].stateMachineProperties[stateMachine][id];
+            if (instance.isFinal) {
+                return null;
+            }
+            return instance.stateMachineRef;
+        }
+    };
+};
+
+const mapDispatchToProps = (dispatch): TransitionPropertiesCallbackProps => {
+    return {
+        setPrivateTopic: (privateSend) => {
+            dispatch(setPrivateTopic(privateSend));
+        },
+        setCurrentId: (id) => {
+            dispatch(setCurrentId(id));
+        },
+        setJsonMessageString: (jsonMessageString) => {
+            dispatch(setJsonMessageString(jsonMessageString));
+        },
+        hideTransitionProperties: () => {
+            dispatch(hideTransitionProperties());
+        },
+        send: (component, stateMachine, messageType, jsonMessageString) => {
+            sessionXCSpy.getPromiseCreateSession()
```

Comment:
same comment: no xcomponent call in the ui (you have to separate ui and logic layers)